Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Model/Building/SqlStorageTypeInformationProvider.cs

```diff
@@ -124,6 +125,10 @@ protected virtual StorageTypeInformation GetStorageType (PropertyDefinition prop
 
       var dotNetType = propertyDefinition.PropertyType;
       var isNullableInDatabase = propertyDefinition.IsNullable || forceNullable;
+
+      if (TryGetDateTimeStorageTypeInformation(propertyDefinition, isNullableInDatabase, dotNetType, out var dateTimeStorageType))
```

Comment:
I believe the tests will show this anyway: we have an issue with nullablity now. 

Design: I think it would be better if we pass the Property.Defintion.PropertyInformation into GetStorageType and evaluate there. The PropertyInfo would need to be nullable. Alternatively, we could also pass only a Func for looking up the attribute. The func feels a bit weird, though.